lib/block_view_dao_coin_limit_order.go

```diff
@@ -37,37 +38,135 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
-	// Validate OperationType is one of our supported enum values
-	// Validate PriceNanos > 0
-	//   If denominated in DESO, confirm PriceNanos is uint64
-	//   PriceNanos == uint256.ToUint64(PriceNanos).ToUint256()
-	// Validate Quantity > 0
-	// If DESO buy, validate quantity < max uint64, order.Quantity.IsUint64()
-	// If DESO buy, validate price * quantity < max uint64
+	// Validate quantity > 0.
+	if !txMeta.Quantity.Gt(uint256.NewInt()) {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+	}
+
+	// Validate quantity is uint64.
+	if !txMeta.Quantity.IsUint64() {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+	}
+
+	// If $DESO buy, validate price * quantity < max uint64.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+		if txMeta.PriceNanos.Uint64()*txMeta.Quantity.Uint64() >= math.MaxUint64 {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+		}
+	}
+
 	// If DAO coin buy, validate price * quantity < max uint256
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin {
+		if !MaxUint256.Gt(uint256.NewInt().Mul(&txMeta.PriceNanos, &txMeta.Quantity)) {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+		}
+	}
 
-	// Validate transfer restriction status, if Dao coin can only be transferred to whitelisted members
+	// Validate transfer restriction status, if Dao coin can only be transferred to whitelisted members.
+	// TODO
 
-	// Validate that buyer has enough $ to buy the DAO coin
-	// Validate that the seller has the DAO coin they're selling
+	// Validate if ask order, that the seller has enough of the DAO coin they're trying to sell.
+	if txMeta.OperationType == DAOCoinLimitOrderEntryOrderTypeAsk {
+		balanceEntry := bav._getBalanceEntryForHODLerPKIDAndCreatorPKID(transactorPKID, txMeta.DAOCoinCreatorPKID, true)
 
-	// Validate that txn specifies inputs to cover $DESO spent on DAO coins
+		// User is trying to open an ask order but doesn't have any of the promised DAO coins.
+		if balanceEntry == nil || balanceEntry.isDeleted {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInsufficientDAOCoinsToOpenAskOrder
+		}
+
+		// User is trying to open an ask order but doesn't have enough of the promised DAO coins.
+		if balanceEntry.BalanceNanos.Lt(&txMeta.Quantity) {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInsufficientDAOCoinsToOpenAskOrder
+		}
+	}
 
-	// ------ End validations
+	// Validate if bid order, that buyer has enough $ to buy the DAO coin.
+	if txMeta.OperationType == DAOCoinLimitOrderEntryOrderTypeBid {
+		if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+			desoBalanceNanos, err := bav.GetDeSoBalanceNanosForPublicKey(bav.GetPublicKeyForPKID(transactorPKID))
+
+			if err != nil {
+				return 0, 0, nil, err
+			}
+
+			// User is trying to open a bid order but doesn't have enough $DESO.
+			if desoBalanceNanos < (txMeta.PriceNanos.Uint64() * txMeta.Quantity.Uint64()) {
```

Comment:
I think you'll want to make a new Uint256 out of the desoBalanceNanos and compare as uint256s to avoid overflow issues